### PR TITLE
Consolidate Ad Creation

### LINKED ISSUE #2535: Consolidate the ad creation step in the Ads Setup flow with the one used in Onboarding
Part of https://github.com/woocommerce/google-listings-and-ads/issues/2460

When a merchant is setting up a new ads campaign from the plugin dashboard, they enter the "Set up paid campaign" flow, which includes individual steps for connecting a Google Ads account, configuring a campaign, and completing their billing setup.

image

Currently, the second step is where a merchant configures their campaign, including setting their targeted audience(s) and their daily budget.

We have a similar form on the last step of onboarding which is being simplified as part of #2458. 

image

Maintaining two different components for this same action is redundant so we will consolidate on a single component that is used in both flows. However, the form displayed in the "Set up paid campaign" flow is also used when editing current campaigns, so the ability to set your audience will need to be optionally shown if we're editing a campaign.

 

## Acceptance Criteria

- [ ] When a user is on the "Create your paid campaign" step of the "Set up paid ads" flow, they will see the same form used on the "Create a campaign" step of the "Get started..." onboarding flow.
- [ ] The merchant can set their daily budget for their campaign
- [ ] If a merchant hasn't set up billing for their Ads account, they will not be able to complete the form and the button will be disabled.
- [ ] If a merchant hasn't set up billing, show the billing setup component
- [ ] A user should be able to successfully set up billing from this form
- [ ] Once billing is set up, the completion button should be enabled and the clicking should successfully create a campaign
- [ ] When editing a campaign, the user should no longer see target audience fields, following #2501.
- [ ] ~When editing a campaign, the user should be able to modify the target audiences~ (no longer applicable)

## Implementation Brief

The PR should get merged into the [feature/2460-simplify-paid-ads-setup](https://github.com/woocommerce/google-listings-and-ads/tree/feature/2460-simplify-paid-ads-setup) branch.

There is a generic component for setting up an ads campaign in `js/src/components/paid-ads/ads-campaign.js`, which returns a `StepContent` component. This component is used for both the `CreatePaidAdsCampaign` and `EditPaidAdsCampaign` pages. This component should be refactored to include the sections from the `SetupPaidAds` component in `js/src/setup-mc/setup-stepper/setup-paid-ads/setup-paid-ads.js` which is used in the onboarding flow, and that component should now consume the generic `AdsCampaign` component.

The generic component should include the ability to pass a title and description to the `StepContentHeader` as props. It should render the `PaidAdsFeaturesSection` and `PaidAdsSetupSections` components from `js/src/setup-mc/setup-stepper/setup-paid-ads/`, both of which should be moved to the `js/src/components/paid-ads/ads-campaign/` folder. The generic component should support configuring two CTA buttons: a primary button that is required, and a secondary button which is optional (for the "Skip this step for now" CTA).

We removed the Audience field from the onboarding flow in #2551, but it will need to be retained as a field that is only shown in the consolidated flow if we're editing a campaign. This will likely require a new boolean `isEditing` prop for the `AdsCampaign` component that can used to determine whether this field (and potentially others in the future) are shown.

## Test Coverage
- E2E tests for tests/e2e/specs/setup-mc/step-4-complete-campaign.test.js should still pass without needing any functional changes.
- E2E tests for tests/e2e/specs/add-paid-campaigns/add-paid-campaigns.test.js should be updated to account for the changes in this issue.

## Definition Questions
- We've removed the Ads audience field from the onboarding form in [#2501](https://github.com/woocommerce/google-listings-and-ads/issues/2501). Do we need to add it back as an optionally displayed field so it can be shown in the secondary flow?
- The generic `AdsCampaign` component used in the paid ads-stepper is also used in pages for creating a campaign and editing a campaign. How are those two pages tested?

### PR DESCRIPTION
**Pull request by asvinb**
### Changes proposed in this Pull Request:

Closes #2535 

_Replace this with a good description of your changes & reasoning._

### Detailed test instructions:

1. 
2. 
3. 

### Changelog entry

>

### CODE REVIEW

File: js/src/components/paid-ads/ads-campaign/index.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+			) }
+
+			{ shouldShowPaidAdsSetup && (
+				<PaidAdsSetupSections
+					onStatesReceived={ handleOnStatesReceived }
+					campaign={ campaign }
+				/>
+			) }
+
+			<FaqsSection />
```
Comment (joemcgill) [inline js/src/components/paid-ads/ads-campaign/index.js:162]: The changes to the AdsCampaign component in #2531 need to be preserved here.

File: js/src/components/paid-ads/ads-campaign/index.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+		setCompleting( event.target.dataset.action );
+
+		onContinue( paidAds );
+	};
+
+	// The status check of Google Ads account connection is included in `paidAds.isReady`,
+	// because when there is no connected account, it will disable the budget section and set the `amount` to `undefined`.
+	const disabledComplete =
+		completing === ACTION_SKIP || ! paidAds.isReady || ! isValidForm;
+	const shouldShowPaidAdsSetup = ! onboardingSetup || showPaidAdsSetup;
```
Comment (joemcgill) [inline js/src/components/paid-ads/ads-campaign/index.js:103]: The shouldShowPaidAdsSetup logic was removed in #2533. Is there a reason it needs to be added back here?

**Review by joemcgill — COMMENTED** [review]
@asvinb wanted to leave an early review with some observations I'm seeing. I can see that you have copied the onboarding flow's component, `SetupPaidAds`, to a new `AdsCampaign` component, which makes sense to me. However, in doing so, it doesn't include the changes made to that form related to [#2459](https://github.com/woocommerce/google-listings-and-ads/issues/2459). I worry that this is going to lead to some challenging merge conflict resolution. Is it possible to try merging the [feature/2459-campaign-creation-flow](https://github.com/woocommerce/google-listings-and-ads/tree/feature/2459-campaign-creation-flow) branch into this branch and ensure those changes are all included?

Additionally, to try to reduce some of the complexity juggling work in parallel with tickets like this one, #2502, and #2503, we should merge all of the PRs targeted to [feature/2460-simplify-paid-ads-setup](https://github.com/woocommerce/google-listings-and-ads/tree/feature/2460-simplify-paid-ads-setup) to the feature branch for #2459 and make them one feature rather than two.

**Comment by asvinb** [comment]
@joemcgill The PR is no longer relevant unfortunately. I'll close it in favour of https://github.com/woocommerce/google-listings-and-ads/pull/2623

**Pull request closed without merging**